Bump minimum CMake version

lsf37 opened the pull request:
See seL4 https://github.com/seL4/seL4/commit/e7bb62d2 for rationale.

lsf37 committed:
Bump minimum CMake version

See seL4 https://github.com/seL4/seL4/commit/e7bb62d2 for rationale.

Signed-off-by: Gerwin Klein <gerwin.klein@proofcraft.systems>

Indanz reviewed (approved):
Is it really necessary for all those CMake files to have a minimum CMake version specified, can't whichever CMake file includes them do that instead? I guess not.

lsf37 committed:
trivial: update style to new cmake-format

Signed-off-by: Gerwin Klein <gerwin.klein@proofcraft.systems>

lsf37 committed:
trivial: remove obsolete license files

These are superseded by the corresponding files in LICENSES and the SPDX
tags.

Signed-off-by: Gerwin Klein <gerwin.klein@proofcraft.systems>

lsf37 commented:
Have updated the cmake style of the files that are touched, but am leaving out the vendored external libraries, so I'm expecting the style checker to still complain about those. That should not be a problem any more in future PRs, though, since we're not planning to change them.
